src/interface/viewport.c

```diff
@@ -2243,7 +2257,10 @@ rct_xy16 screen_coord_to_viewport_coord(rct_viewport *viewport, uint16 x, uint16
 rct_xy16 viewport_coord_to_map_coord(int x, int y, int z)
 {
 	rct_xy16 ret;
-	switch (RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_ROTATION, uint32)) {
+	uint32 rotation = RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_ROTATION, uint32);
+	switch (rotation) {
+	default:
+		log_warning("Improper rotation %u", rotation);
```

Comment:
I know there is a wrong direction somewhere, even though I don't recall exactly where and how to reproduce it. I'd expect a wrong `rotation` can be introduced just as easily.

Ideally, I'd like all of the code to handle such cases gracefully and I've seen you did `direction & 3` somewhere recently, which is almost exactly what I'm after, minus the warning message.

I'd be happy to make it optional at compile time, possibly a `check` macro (as I don't think assert on such a thing is a good experience)